platform/ui/src/components/downloadDialog/DownloadDialog.js

```diff
@@ -0,0 +1,235 @@
+import './DownloadDialog.styl';
+
+import React, { PureComponent } from 'react';
+import { withTranslation } from '../../utils/LanguageProvider';
+import Modal from 'react-bootstrap-modal';
+import 'react-bootstrap-modal/lib/css/rbm-patch.css';
+import PropTypes from 'prop-types';
+
+class DownloadDialog extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: this.props.isOpen,
+      width: 3000,
+      height: 3000,
+      fileName: '',
+      fileType: 'png',
+      showAnnotations: true,
+      fileTypeOptions: [
+        {
+          key: 'jpg',
+          value: 'jpg'
+        },
+        {
+          key: 'png',
+          value: 'png'
+        }
+      ],
+      previewElementRef: null
+    };
+
+    this.submitForm = this.submitForm.bind(this);
+    this.onClose = this.onClose.bind(this);
+  }
+
+  submitForm(e) {
+    e.preventDefault();
+
+    const formData = {
+      width: this.state.width,
+      height: this.state.height,
+      fileName: this.state.fileName,
+      fileType: this.state.fileType,
+      showAnnotations: this.state.showAnnotations,
+    };
+
+    const thereIsAnyEmpty = Object.values(formData).some(field => field === '');
```

Comment:
I'm not sure we need validation on this form. It should be "safe enough" to fallback to defaults if no value is provided. I'm not sure what value validation would provide, and it could actually hamper the experience.

Reply:
Validation updated to avoid empties and bad scaled sizes.